be2-scala/src/test/scala/ch/epfl/pop/json/HighLevelProtocolSuite.scala

```diff
@@ -126,6 +130,48 @@ class HighLevelProtocolSuite extends FunSuite with Matchers {
         an[IllegalArgumentException] should be thrownBy Message.buildFromJson(source)
     }
   }
+
+  test("parse correctly heartbeat") {
+    val heartbeat = "{\n    \"jsonrpc\": \"2.0\",\n    \"method\": \"heartbeat\",\n    \"params\": {\n        \"/root/nLghr9_P406lfkMjaNWqyohLxOiGlQee8zad4qAfj18=/social/8qlv4aUT5-tBodKp4RszY284CFYVaoDZK6XKiw9isSw=\": [\n            \"DCBX48EuNO6q-Sr42ONqsj7opKiNeXyRzrjqTbZ_aMI=\"\n        ],\n        \"/root/nLghr9_P406lfkMjaNWqyohLxOiGlQee8zad4qAfj18=/HnXDyvSSron676Icmvcjk5zXvGLkPJ1fVOaWOxItzBE=\": [\n            \"z6SbjJ0Hw36k8L09-GVRq4PNmi06yQX4e8aZRSbUDwc=\",\n            \"txbTmVMwCDkZdoaAiEYfAKozVizZzkeMkeOlzq5qMlg=\"\n        ]\n    }\n}"
```

Comment:
A good idea would be to add a function which directly returns the pretty print version of the json methods found in the resources file.
You can find a similar function, `getExampleMessage`  in the `MessageDataProtocolSuite.scala`.

Reply:
I got my way out of writing a pretty print function. I can call `HighLevelProtocol.write()` to get a `spray.json` and use that library's pretty print function. Then I read from that string with `JsonRpcRequest.buildFromJson` and check if every field is what's expected. I added some more tests to get a better coverage but I don't think I can go much higher than that, most of the coverage missing is from the renaming of `ParamsHandler` to `ParamsWithChannelHandler` for which I added tests too.